src/validation.cpp

```diff
@@ -1813,9 +1813,9 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
         //  relative to a piece of software is an objective fact these defaults can be easily reviewed.
         // This setting doesn't force the selection of any particular chain but makes validating some faster by
         //  effectively caching the result of part of the verification.
-        BlockMap::const_iterator  it = mapBlockIndex.find(hashAssumeValid);
-        if (it != mapBlockIndex.end()) {
-            if (it->second->GetAncestor(pindex->nHeight) == pindex &&
+        const CBlockIndex* pindex_valid = LookupBlockIndex(hashAssumeValid);
```

Comment:
Dunno if its worth bothering for things in CChainState. The whole goal here is to make mapBlockIndex a private in CChainState and expose LookupBlockIndex externally.

Reply:
I'd prefer to avoid special cases for now. Eventually `LookupBlockIndex` will loose `inline`, so do it then?

Reply:
For cases where we are fine with a const CBlockIndex* result, I dont think it matters much, but I'd kinda like to have LookupBlockIndex only return const CBlockIndex*es, so for things like CChainState::AddToBlockIndex, we'd have to partially revert this change then, easier to not do it in the first place IMO.

Reply:
> have LookupBlockIndex only return const CBlockIndex*es

If you look closely you will see that a lot of function signatures require non-const `CBlockIndex*`, although a const should do the work (almost?) everywhere. I'm happy to do that in a follow up and deal with the non-const cases there. At least this PR is *complete*.

Reply:
Making everything const isnt so hard, see #10692, which did it mostly in scripted-diffs.